scripts/src/java/org/oppia/android/scripts/testfile/TestFileCheck.kt

```diff
@@ -26,7 +26,9 @@ fun main(vararg args: String) {
   // A list of all the files to be exempted for this check.
   // TODO(#3436): Develop a mechanism for permanently exempting files which do not ever need tests.
   val testFileExemptionList = loadTestFileExemptionsProto(testFileExemptiontextProto)
-    .getExemptedFilePathList()
+    .testFileExemptionList
+    .filter { it.testFileNotRequired }
+    .map { it.exemptedFilePath }
```

Comment:
Can you also add a test in ``TestFileCheckTest`` to verify that a min coverage exemption results in the file _not_ being exempted here, as expected? We aren't going to verify beyond that in ``TestFileCheck`` so it seems useful to verify here.

Reply:
@BenHenning, I would like to confirm the approach that I can take to write the test for this case.
So the existing cases use the `main` call to execute the tests and that internally uses the actual test_file_exemption.textproto values to check the test exemptions

1. In our case to check that the `override_min_coverage_percentage` exempted tests don't interfere with the actual test file checks we might need to have the actual data inside, so would we need to add atleast one `override_min_coverage_percentage` to the actual "test_file_exemptions.textproto" to facilitate this check as done with other test cases or

2. As the entire logic is in `main`, should it be separate to appropriate classes and functions (**eg:** Class `TestFileCheck`, `fun getTestFileExemptions`) so we can call it passing our own "testing_test_file_exemptions.textproto" that includes a `override_min_coverage_percentage` value and use that for testing.

or is there any other much straight forward way to do that?

Reply:
So Ben, I refactored the `TestFileCheck` to separate the logic from `main` primarily to have the ability to pass in a custom protobuf (as path) to check the `override_min_coverage_percentage` exempted file is not being exempted from the actual test file check. 

Now it has 2 functions:
1. **`getTestFileExemptionList`** 
   that gets the path of the test file exemption textproto file and returns a list of file paths that are exempted from requiring a test file.
3. **`execute`** 
  that actually executes the test file check.

Utilizing this, I created the test `testTestFileCheck_minCoverageExemptedFile_notExemptedFromTestFileCheck` that sends a custom `proto` we built 
```
test_file_exemption {
  exempted_file_path: "test/TestExemptedFile.kt"
  test_file_not_required: true
}
test_file_exemption {
  exempted_file_path: "coverage/CoverageTestFile.kt"
  override_min_coverage_percent_required: 5
}
```
to send to `TestFileCheck(_, _).getTestFileExemptionList(testFileExemptiontextProto)`, which filters and just returns the list of files that need to be exempted.

### Things I tried:
To not split up the `getTestExemptionList` function as it may result in making it public, but that is necessary or else it will still use the old `protobuf`. So, this, for now, breaks the part of over-exposing, but I guess that is needed here.

Reply:
It shouldn't actually be needed to be exposed--I left a few other comments for considerations that, when taken together, should mean that we don't need to approach the test part quite this way.

Resolving this comment, though, since a test was added.